### PR TITLE
7 maven archetypes

### LINKED ISSUE #7: create Maven Archetype
In relation to #6, create a [Maven Archetype](https://maven.apache.org/guides/introduction/introduction-to-archetypes.html) to help generate sample applications using Ahab.

### PR DESCRIPTION
**Pull request by hinchliff**
This maven archetype will generate a new maven project, that is ready to compile and run.  When run, it will create a simple slice using Ahab.

@paul-ruth is this a reasonable example project?  Does it demonstrate correct use of Ahab?

Fixes #7

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,7 +15,8 @@ Use the following pom snippet to include Ahab in your project:
 <dependency>
     <groupId>org.renci.ahab</groupId>
     <artifactId>ahab</artifactId>
-    <version>0.1.2</version>
+    <version>0.1.6</version>
```
Comment (mjstealey) [inline README.md:18]: @hinchliff - Other code references `<version>0.1.7-SNAPSHOT</version>`, should this be consistent? or nah?

Comment (hinchliff) [reply README.md:18]: well, 0.1.6 is the current release...  and it doesn't make sense to have the README reference a snapshot version.
but the README lagging the latest release is a problem that i don't really know a good (simple) solution too.

File: ahab-maven-archetypes/pom.xml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>org.renci.ahab</groupId>
+        <artifactId>ahab-project</artifactId>
+        <version>0.1.7-SNAPSHOT</version>
```
Comment (mjstealey) [inline ahab-maven-archetypes/pom.xml:9]: Should this then reference 0.1.6?

File: ahab-maven-archetypes/ahab-simple-archetype/pom.xml [hunk 3]
```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0"?>
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd"
+         xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>org.renci.ahab</groupId>
+        <artifactId>ahab-maven-archetypes</artifactId>
+        <version>0.1.7-SNAPSHOT</version>
```
Comment (mjstealey) [inline ahab-maven-archetypes/ahab-simple-archetype/pom.xml:9]: @hinchliff - 0.1.6? or snapshot?

Comment (hinchliff) [reply ahab-maven-archetypes/ahab-simple-archetype/pom.xml:9]: the versions in the poms will automatically get updated by Maven.

Comment (mjstealey) [reply ahab-maven-archetypes/ahab-simple-archetype/pom.xml:9]: Ah, yes, maven, this is what I get for reviewing Java code while having a Python project open in my IDE...

**Review by mjstealey — APPROVED** [review]
Looks good to me and I look forward to making use of this as I go down the documentation rabbit hole.

**Comment by ibaldin** [comment]
@paul-ruth why hasn't this been pulled in?

**Pull request merged**